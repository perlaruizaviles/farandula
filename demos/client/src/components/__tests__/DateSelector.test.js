import React from 'react';
import {shallow} from 'enzyme';
import DateSelector from '../Common/DateSelector';
import moment from 'moment';

describe('Rendering DateSelector', () => {

  const now = moment("2017-05-22T13:48:18.677");
  
  const oneYearFromNow = moment("2018-05-22T13:48:18.677");

  const props = {
    minDate: now,
    maxDate: oneYearFromNow,
    startDate: now,
    endDate: now,
    selected: now,
    changeTravelDate: () => {}
  };

  const tree = shallow(<DateSelector {...props} />);

  it('Should create an snapshot for DateSelector', () => {
    expect(tree).toMatchSnapshot();
  });

  it('Renders DateSelector', () => {
    expect(tree.find('t').length).toBe(1);
  });

  it('Min date', () => {
    expect(tree.find('t').props().minDate).toBe(now);
  });

  it('Max date', () => {
    expect(tree.find('t').props().maxDate).toBe(oneYearFromNow);
  });
});
