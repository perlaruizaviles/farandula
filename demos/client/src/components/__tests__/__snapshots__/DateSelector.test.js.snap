exports[`Rendering DateSelector Should create an snapshot for DateSelector 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <t
    customInput={
        <Input
            icon="calendar outline"
            style={
                Object {
                    "color": "#216ba5",
                    "width": "150px",
                  }
            }
            type="text" />
    }
    dateFormat="L"
    dateFormatCalendar="MMMM YYYY"
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="scroll"
    endDate={"2017-05-22T20:48:18.677Z"}
    maxDate={"2018-05-22T20:48:18.677Z"}
    minDate={"2017-05-22T20:48:18.677Z"}
    monthsShown={1}
    onBlur={[Function]}
    onChange={[Function]}
    onClickOutside={[Function]}
    onFocus={[Function]}
    onMonthChange={[Function]}
    onSelect={[Function]}
    placeholderText="Select date..."
    popoverAttachment="top left"
    popoverTargetAttachment="bottom left"
    popoverTargetOffset="10px 0"
    selected={"2017-05-22T20:48:18.677Z"}
    startDate={"2017-05-22T20:48:18.677Z"}
    tetherConstraints={
        Array [
            Object {
              "attachment": "together",
              "to": "window",
            },
          ]
    }
    utcOffset={-420}
    withPortal={false} />,
  "nodes": Array [
    <t
      customInput={
            <Input
                  icon="calendar outline"
                  style={
                        Object {
                              "color": "#216ba5",
                              "width": "150px",
                            }
                  }
                  type="text" />
      }
      dateFormat="L"
      dateFormatCalendar="MMMM YYYY"
      disabled={false}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      endDate={"2017-05-22T20:48:18.677Z"}
      maxDate={"2018-05-22T20:48:18.677Z"}
      minDate={"2017-05-22T20:48:18.677Z"}
      monthsShown={1}
      onBlur={[Function]}
      onChange={[Function]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      placeholderText="Select date..."
      popoverAttachment="top left"
      popoverTargetAttachment="bottom left"
      popoverTargetOffset="10px 0"
      selected={"2017-05-22T20:48:18.677Z"}
      startDate={"2017-05-22T20:48:18.677Z"}
      tetherConstraints={
            Array [
                  Object {
                    "attachment": "together",
                    "to": "window",
                  },
                ]
      }
      utcOffset={-420}
      withPortal={false} />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <DateSelector
        changeTravelDate={[Function]}
        endDate={"2017-05-22T20:48:18.677Z"}
        maxDate={"2018-05-22T20:48:18.677Z"}
        minDate={"2017-05-22T20:48:18.677Z"}
        selected={"2017-05-22T20:48:18.677Z"}
        startDate={"2017-05-22T20:48:18.677Z"} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "changeTravelDate": [Function],
          "endDate": "2017-05-22T20:48:18.677Z",
          "maxDate": "2018-05-22T20:48:18.677Z",
          "minDate": "2017-05-22T20:48:18.677Z",
          "selected": "2017-05-22T20:48:18.677Z",
          "startDate": "2017-05-22T20:48:18.677Z",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <t
          customInput={
                    <Input
                              icon="calendar outline"
                              style={
                                        Object {
                                                  "color": "#216ba5",
                                                  "width": "150px",
                                                }
                              }
                              type="text" />
          }
          dateFormat="L"
          dateFormatCalendar="MMMM YYYY"
          disabled={false}
          disabledKeyboardNavigation={false}
          dropdownMode="scroll"
          endDate={"2017-05-22T20:48:18.677Z"}
          maxDate={"2018-05-22T20:48:18.677Z"}
          minDate={"2017-05-22T20:48:18.677Z"}
          monthsShown={1}
          onBlur={[Function]}
          onChange={[Function]}
          onClickOutside={[Function]}
          onFocus={[Function]}
          onMonthChange={[Function]}
          onSelect={[Function]}
          placeholderText="Select date..."
          popoverAttachment="top left"
          popoverTargetAttachment="bottom left"
          popoverTargetOffset="10px 0"
          selected={"2017-05-22T20:48:18.677Z"}
          startDate={"2017-05-22T20:48:18.677Z"}
          tetherConstraints={
                    Array [
                              Object {
                                "attachment": "together",
                                "to": "window",
                              },
                            ]
          }
          utcOffset={-420}
          withPortal={false} />,
        "_debugID": 2,
        "_renderedOutput": <t
          customInput={
                    <Input
                              icon="calendar outline"
                              style={
                                        Object {
                                                  "color": "#216ba5",
                                                  "width": "150px",
                                                }
                              }
                              type="text" />
          }
          dateFormat="L"
          dateFormatCalendar="MMMM YYYY"
          disabled={false}
          disabledKeyboardNavigation={false}
          dropdownMode="scroll"
          endDate={"2017-05-22T20:48:18.677Z"}
          maxDate={"2018-05-22T20:48:18.677Z"}
          minDate={"2017-05-22T20:48:18.677Z"}
          monthsShown={1}
          onBlur={[Function]}
          onChange={[Function]}
          onClickOutside={[Function]}
          onFocus={[Function]}
          onMonthChange={[Function]}
          onSelect={[Function]}
          placeholderText="Select date..."
          popoverAttachment="top left"
          popoverTargetAttachment="bottom left"
          popoverTargetOffset="10px 0"
          selected={"2017-05-22T20:48:18.677Z"}
          startDate={"2017-05-22T20:48:18.677Z"}
          tetherConstraints={
                    Array [
                              Object {
                                "attachment": "together",
                                "to": "window",
                              },
                            ]
          }
          utcOffset={-420}
          withPortal={false} />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <DateSelector
    changeTravelDate={[Function]}
    endDate={"2017-05-22T20:48:18.677Z"}
    maxDate={"2018-05-22T20:48:18.677Z"}
    minDate={"2017-05-22T20:48:18.677Z"}
    selected={"2017-05-22T20:48:18.677Z"}
    startDate={"2017-05-22T20:48:18.677Z"} />,
}
`;
